controllers/managedcluster/managedcluster-controller.go

```diff
@@ -79,8 +79,8 @@ func (r *ManagedClusterReconciler) Reconcile(ctx context.Context, request ctrl.R
 		return ctrl.Result{}, err
 	}
 
-	// annotation format is <name>.<namespace>.<kind>.<apiversion>
-	expectedProvisioner := fmt.Sprintf("%s.%s.%s.%s", claimName, claimNamespace, clusterClaim.Kind, clusterClaim.APIVersion)
+	// annotation format is <name>.<namespace>
```

Comment:
@xiangjingli the kind and apiversion should not be removed, I guess this is the root cause for https://issues.redhat.com/browse/ACM-16131